tests/functional/adapter/test_seeds_column_types_overrides.py

```diff
@@ -0,0 +1,139 @@
+        "smallint_example": "smallint",
+        "bigint_example": "bigint",
+        "real_example": "real",
+        "double_example": "double",
+        "decimal_example": "decimal",
+        "decimal_p_example": "decimal(3,2)",
+    },
+    "string_type": {
+        "varchar_example": "varchar",
+        "varchar_n_example": "varchar(10)",
+        "char_example": "char",
+        "char_n_example": "char(10)",
+        "varbinary_example": "varbinary",
+        "json_example": "json",
+    },
+}
+
+
+# function copied from dbt.tests.util. Original function doesn't return numeric_precision and numeric_scale.
+def get_relation_columns(adapter, name):
+    relation = relation_from_name(adapter, name)
+    with get_connection(adapter):
+        columns = adapter.get_columns_in_relation(relation)
+        return sorted(
+            (
+                (c.name, c.dtype, c.char_size, c.numeric_precision, c.numeric_scale)
+                for c in columns
+            ),
+            key=lambda x: x[0],
+        )
+
+
+class TestSeedsColumnTypesOverrides:
+    @property
+    def schema(self):
+        return "default"
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "seeds": {
+                "test": {
+                    "boolean_type": {"+column_types": seed_types["boolean_type"]},
+                    "datetime_type": {"+column_types": seed_types["datetime_type"]},
+                    "number_type": {"+column_types": seed_types["number_type"]},
+                    "string_type": {"+column_types": seed_types["string_type"]},
+                }
+            }
+        }
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "boolean_type.csv": boolean_type,
+            "datetime_type.csv": datetime_type,
+            "number_type.csv": number_type,
+            "string_type.csv": string_type,
+        }
+
+    def test_seeds_column_overrides(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 4
+
+        for seed in seed_types:
+            # retrieve information about columns from trino
+            columns = get_relation_columns(project.adapter, seed)
+            for column in columns:
+                # if precision/scale/char_length was explicitly specified in column_types config,
+                # compare specified and retrieved column's precision/scale/char_length,
+                # then compare specified and retrieved column type
+                if "(" in seed_types[seed][column[0]]:
+                    # if precision/char_length is specified:
+                    if "," not in seed_types[seed][column[0]]:
+                        split = re.split(r"[()]", seed_types[seed][column[0]])
+                        # precision/char_length:
+                        assert split.pop(1) == str(column[2])
```

Comment:
Please use destructuring instead of `pop`. See https://blog.teclado.com/destructuring-in-python/

Reply:
changed